content/modeling/modeling_3.md

```diff
@@ -0,0 +1,117 @@
+
+If you wish to change this value, copy the `conversion_value.sql` file from the macros folder in the snowplow_fractribution package (at `[dbt_project_name]/dbt_packages/snowplow_fractribution/macros/conversion_value.sql`) and add it to the macros folder of your own dbt project. Update the value and save the file.
+
+**Configure the default channel_classification macro**
+
+The channel_classification macro is used to perform channel classifications. This can be altered to generate your expected channels if they differ from the channels generated in the default macro. It is highly recommended that you examine and configure this macro when using your own data, as the default values will not consider any custom marketing parameters.
+
+If you wish to change the channel classification macro, copy the `channel_classification.sql` file from the macros folder in the snowplow_fractribution package (at `[dbt_project_name]/dbt_packages/snowplow_fractribution/macros/channel_classification.sql`) and add it to the macros folder of your own dbt project. Update the SQL and save the file.
+ 
+***
+
+#### **Step 3:** Run the model
+
+Execute the following either through your CLI or from within dbt Cloud
+
+```
+dbt run --select snowplow_fractribution
+```
+
+This should take a couple of minutes to run.
+
+***
+
+#### **Step 4:** Check the output schema
+Head to the SQL editor of your choice (e.g.: Snowflake Web UI) to check the model's output. You should be able to see the data under the _derived schema.
+
+***
+
+#### **Step 5:** Explore the data created by your dbt models
+
+Take some time to familiarise yourself with the derived tables. These tables are used in the next step to fractionally attribute revenue to channels. Tables output by the snowplow_fractribution dbt package are:
+
+- `snowplow_fractribution_channel_counts`: Number of events grouped by channel, campaign, source and medium.
+- `snowplow_fractribution_channel_spend`: Spend on each channel, used in ROAS calculations.
+- `snowplow_fractribution_conversions_by_customer_id`: Conversion revenue for each conversion, along with the associated customerid.
+- `snowplow_fractribution_path_summary`: Summary of different path combinations and associated conversion/non-conversions.
+- `snowplow_fractribution_paths_to_conversion`: Path combinations leading to conversion.
+- `snowplow_fractribution_paths_to_non_conversion`: Path combinations leading to non-conversion.
+- `snowplow_fractribution_sessions_by_customer_id`: Channel information by session timestamp, where an event timestamp is considered as the session start.
+
+***
+
+Next we will run a Python script to create the actual Fractribution table.
```

Comment:
I have tried to run all of the attribution models and two of them won't run due to the same error (first_touch, last_touch): `TypeError: field Referral: Cannot merge type <class 'snowflake.snowpark.types.LongType'> and <class 'snowflake.snowpark.types.FloatType'>` It happens when it tries to create a snowpack dataframe. I have only just discovered it so I do not have time to investigate further today, but hopefully you can recreate the issue as I was using the test data . If not I will have a look tomorrow.

Reply:
Thanks, I missed this. Mike and I fixed this today, should work as expected now.